dsls/sql/src/main/java/org/apache/beam/dsls/sql/transform/BeamAggregationTransforms.java

```diff
@@ -134,545 +134,201 @@ public Instant apply(BeamSqlRow input) {
-
-        hasCount = true;
-        countIndex = aggDataType.size() - 1;
-      }
-
-      this.aggDataType = BeamSqlRecordType.create(fieldNames, fieldTypes);
-      this.countIndex = countIndex;
-    }
-
-    private void verifySupportedAggregation(AggregateCall ac) {
-      //donot support DISTINCT
-      if (ac.isDistinct()) {
-        throw new UnsupportedOperationException("DISTINCT is not supported yet.");
-      }
-      String aggFnName = ac.getAggregation().getName();
-      switch (aggFnName) {
-      case "COUNT":
-        //COUNT works for any data type;
-        break;
-      case "SUM":
-        // SUM only support for INT, LONG, FLOAT, DOUBLE, DECIMAL, SMALLINT,
-        // TINYINT now
-        if (!Arrays
-            .asList(SqlTypeName.INTEGER, SqlTypeName.BIGINT, SqlTypeName.FLOAT, SqlTypeName.DOUBLE,
-                SqlTypeName.SMALLINT, SqlTypeName.TINYINT)
-            .contains(ac.type.getSqlTypeName())) {
-          throw new UnsupportedOperationException(
-              "SUM only support for INT, LONG, FLOAT, DOUBLE, SMALLINT, TINYINT");
-        }
-        break;
-      case "MAX":
-      case "MIN":
-        // MAX/MIN only support for INT, LONG, FLOAT, DOUBLE, DECIMAL, SMALLINT,
-        // TINYINT, TIMESTAMP now
-        if (!Arrays.asList(SqlTypeName.INTEGER, SqlTypeName.BIGINT, SqlTypeName.FLOAT,
-            SqlTypeName.DOUBLE, SqlTypeName.SMALLINT, SqlTypeName.TINYINT,
-            SqlTypeName.TIMESTAMP).contains(ac.type.getSqlTypeName())) {
-          throw new UnsupportedOperationException("MAX/MIN only support for INT, LONG, FLOAT,"
-              + " DOUBLE, SMALLINT, TINYINT, TIMESTAMP");
-        }
-        break;
-      case "AVG":
-        // AVG only support for INT, LONG, FLOAT, DOUBLE, DECIMAL, SMALLINT,
-        // TINYINT now
-        if (!Arrays
-            .asList(SqlTypeName.INTEGER, SqlTypeName.BIGINT, SqlTypeName.FLOAT, SqlTypeName.DOUBLE,
-                SqlTypeName.SMALLINT, SqlTypeName.TINYINT)
-            .contains(ac.type.getSqlTypeName())) {
-          throw new UnsupportedOperationException(
-              "AVG only support for INT, LONG, FLOAT, DOUBLE, SMALLINT, TINYINT");
-        }
-        break;
-      default:
-        throw new UnsupportedOperationException(
-            String.format("[%s] is not supported.", aggFnName));
-      }
-    }
-
-    @Override
-    public BeamSqlRow createAccumulator() {
-      BeamSqlRow initialRecord = new BeamSqlRow(aggDataType);
-      for (int idx = 0; idx < aggElementExpressions.size(); ++idx) {
-        BeamSqlExpression ex = aggElementExpressions.get(idx);
-        String aggFnName = aggFunctions.get(idx);
-        switch (aggFnName) {
-        case "COUNT":
-          initialRecord.addField(idx, 0L);
-          break;
-        case "AVG":
-        case "SUM":
-          //for both AVG/SUM, a summary value is hold at first.
-          switch (ex.getOutputType()) {
+          switch (call.type.getSqlTypeName()) {
```

Comment:
Can you please pull each of these switch statements on getSqlTypeName into a factory method in the corresponding aggregation (e.g., BeamBuiltinAggregations.Max.create(SqlTypeName typeName))? That will help manage some of the size of this method, and also put the per-type construction logic somewhere reusable.